### PR TITLE
Use AppScalefile as fallback source for IPs

### PR DESCRIPTION
**Pull request by cdonati**
After an upgrade from <= 3.1.0, the user may not have a locations JSON file. This change allows the tools to still check if the ssh key is valid by using the AppScalefile to generate a list of all addresses in a cluster.

### CODE REVIEW

**Comment by cdonati** [comment]
https://ocd.appscale.com:8080/job/Daily%20Build/2866/

File: appscale/tools/appscale.py [hunk 1]
```diff
@@ -337,7 +337,11 @@ def valid_ssh_key(self, config, run_instances_opts):
     if not os.path.exists(ssh_key_location):
       return False
 
-    all_ips = LocalState.get_all_public_ips(keyname)
+    try:
+      all_ips = LocalState.get_all_public_ips(keyname)
+    except BadConfigurationException:
+      # If this is an upgrade from 3.1.0, there may not be a locations JSON.
+      all_ips = set(run_instances_opts.ips.values())
```
Comment (obino) [inline appscale/tools/appscale.py]: Few lines below there is a look and then an SSH connection to all IPs, What happens if these IPs are say wrong (like node-2, node-3...)?

Comment (cdonati) [inline appscale/tools/appscale.py]: This function is only run when 'infrastructure' is not in the AppScalefile. I'm open to suggestions for making it more defensive, though.

Comment (whoarethebritons) [inline appscale/tools/appscale.py]: Would checking to see if "node" is in the ip strings work as a first check? However, what would we do if that was the case...

Comment (cdonati) [inline appscale/tools/appscale.py]: Yes, I can add an assert to make sure that we are using IP addresses. I would guess that we want to fail if we are not.

**Comment by cdonati** [comment]
https://ocd.appscale.com:8080/job/Daily%20Build/2994/

**Pull request merged**